drivers/gpio/gpio_stellaris.c

```diff
@@ -95,42 +101,27 @@ static int gpio_stellaris_configure(struct device *dev, int access_op,
 		return -EINVAL;
 	}
 
-	/* Pin digital enable */
-	sys_set_bit(GPIO_REG_ADDR(base, GPIO_DEN_OFFSET), pin);
-
-	/* Input-0,output-1 */
-	if ((flags & GPIO_DIR_MASK) == GPIO_DIR_IN) {
+	if ((flags & GPIO_INPUT) != 0) {
 		sys_clear_bit(GPIO_REG_ADDR(base, GPIO_DIR_OFFSET), pin);
-	} else {
+		/* Pin digital enable */
+		sys_set_bit(GPIO_REG_ADDR(base, GPIO_DEN_OFFSET), pin);
+	} else if ((flags & GPIO_OUTPUT) != 0) {
+		mm_reg_t mask_addr;
 		sys_set_bit(GPIO_REG_ADDR(base, GPIO_DIR_OFFSET), pin);
```

Comment:
We still should enable output mode only after we set pin initial value since at this point DEN (Digital Enable) bit can already be set. It will happen if the pin was previously configured as input and is being reconfigured as output.